x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/support/StringMatcherTests.java

```diff
@@ -23,6 +26,29 @@ public void testEmptySet() throws Exception {
         }
     }
 
+    public void testMatchAllWildcard() throws Exception {
```

Comment:
This test does not really test the current change. There is no easy way to test it. Theoretically a test should not care about the internals of a method. But I wonder whether we could make an exception to ensure the StringMatcher's predicate is a simple "match all":
* Create a static variable `MATCH_ALL` for `s->true` similar to the existing `MATCH_NOTHING`.
* Have a package private method to expose the `predicate` for StringMatcher.

So that the test can assert the match-all short circuit is working.

Reply:
The test was added because we lacked a test for the `*` case, and we need one to demonstrate that this change hasn't broken anything.

On the performance side, we also don't have tests to ensure that exact matches are handled differently than automaton matches, and that seems much more significant.
I can add a test, but it seems weird to go to the trouble of exposing the internals, just so we can test 3 lines of very obvious code simply because they were added in a separate PR to the rest of the class.